### ISSUE TITLE
Failed to invoke 'RequestFile' due to an error on the server. HubException: Method does not exist.

### ISSUE THREAD

**Issue by NickDarvey** [fsbolero/Templating.HotReload#7]
I just upgraded my app to Bolero v0.13.23 and Bolero.HotReload v0.13.7 and I'm getting 'Failed to invoke 'RequestFile' due to an error on the server. HubException: Method does not exist.' when running my app

## Repro
1. Install the template with `dotnet new -i Bolero.Templates::0.13.5`
2. Create a new app with `dotnet new bolero-app -o MyApp`
3. Run the application

## Expected
Application starts and runs with hot reloading

## Actual
Application starts but soon fails with this error in the console:
```
﻿Hot reload connected!
blazor.webassembly.js:1 Hot reload failed to request file: System.AggregateException: One or more errors occurred. (Failed to invoke 'RequestFile' due to an error on the server. HubException: Method does not exist.) ---> Microsoft.AspNetCore.SignalR.HubException: Failed to invoke 'RequestFile' due to an error on the server. HubException: Method does not exist.
```

Additionally, `Microsoft.AspNetCore.SignalR.HubException` is thrown on the server.

## System info
Bolero.Server 0.13.23
Bolero.HotReload.Server 0.13.7
dotnet SDK 3.1.300

Debugging around that exception shows that [the dispatcher does not have a method](https://github.com/dotnet/aspnetcore/blob/2ad8121efbdddd46920918bddb9ee67e1c4144a9/src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs#L645) named 'RequestFile' registered for that hub.

Breaking [during client connection to the hub](https://github.com/dotnet/aspnetcore/blob/2ad8121efbdddd46920918bddb9ee67e1c4144a9/src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs#L82) I can see that `hub` doesn't appear to have that method when checking via reflection.

```
>hub.GetType().AssemblyQualifiedName
"Bolero.Templating.Server.Impl+HotReloadHub, Bolero.Templating.Server, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null"
>hub.GetType().GetMethods()
{System.Reflection.MethodInfo[13]}
    [0]: {Microsoft.AspNetCore.SignalR.IHubCallerClients get_Clients()}
    [1]: {Void set_Clients(Microsoft.AspNetCore.SignalR.IHubCallerClients)}
    [2]: {Microsoft.AspNetCore.SignalR.HubCallerContext get_Context()}
    [3]: {Void set_Context(Microsoft.AspNetCore.SignalR.HubCallerContext)}
    [4]: {Microsoft.AspNetCore.SignalR.IGroupManager get_Groups()}
    [5]: {Void set_Groups(Microsoft.AspNetCore.SignalR.IGroupManager)}
    [6]: {System.Threading.Tasks.Task OnConnectedAsync()}
    [7]: {System.Threading.Tasks.Task OnDisconnectedAsync(System.Exception)}
    [8]: {Void Dispose()}
    [9]: {System.Type GetType()}
    [10]: {System.String ToString()}
    [11]: {Boolean Equals(System.Object)}
    [12]: {Int32 GetHashCode()}
```

Which I guess explains why the hub dispatcher doesn't see them [when it checks](https://github.com/dotnet/aspnetcore/blob/f17fcfd3c80464d36cf632274ad02f04405efeba/src/SignalR/server/Core/src/Internal/HubReflectionHelper.cs#L17).

I dropped my server assembly into ILSpy and loaded its dependencies.
It shows that Bolero.Templating.Server.dll does have `HotReloadHub` with a base type of `Microsoft.AspNetCore.SignalR.Hub` with a method called `RequestFile` as we'd expect.
```
using Microsoft.FSharp.Control;
using Microsoft.FSharp.Core;
using System;
using System.Threading.Tasks;

[Serializable]
[CompilationMapping(SourceConstructFlags.ObjectType)]
internal class HotReloadHub
{
    internal Watcher watcher;

    public HotReloadHub(Watcher watcher)
        : this()
    {
        this.watcher = watcher;
    }

    internal Task<string> RequestFile(string filename)
    {
        FSharpAsync<string> computation = ExtraTopLevelOperators.DefaultAsyncBuilder.Delay(new RequestFile@85(this, filename));
        return FSharpAsync.StartAsTask(computation, null, null);
    }
}
```

Will continue to dig. Let me know if you have any thoughts.

**Comment by NickDarvey** [comment]
Actually... That decompilation doesn't look quite right. It's marked as internal.

If I break during client connection and fetch non-public methods on the hub type it does actually return `RequestFiles`:
```
>hub.GetType().GetMethods(BindingFlags.Instance | BindingFlags.NonPublic)
{System.Reflection.MethodInfo[4]}
    [0]: {System.Threading.Tasks.Task`1[System.String] RequestFile(System.String)}
    [1]: {Void Dispose(Boolean)}
    [2]: {System.Object MemberwiseClone()}
    [3]: {Void Finalize()}
```

But the [HubReflectionHelper only looks for public methods](https://github.com/dotnet/aspnetcore/blob/f17fcfd3c80464d36cf632274ad02f04405efeba/src/SignalR/server/Core/src/Internal/HubReflectionHelper.cs#L15-L21) with `BindingFlags.Public | BindingFlags.Instance`.

It looks like being marked as internal occurred in commit https://github.com/fsbolero/Templating.HotReload/commit/329567ccf5e68cf8766445cd3cec842d13ea3461

I think that's it, maybe? Would there be any issues making it public again?

**Comment by Tarmil** [comment]
Thanks for the report! This looks like the issue indeed, I'll do some tests and publish a hotfix.

**Comment by Tarmil** [comment]
I could confirm that this was the issue. It is fixed, nuget publish for HotReload 0.13.8 is under way!

**Comment by NickDarvey** [comment]
Thanks, can confirm fixed in 0.13.8.
